### DISCUSSION TITLE
How to achieve some parallel or switch in parent parallel state？

### DISCUSSION THREAD

**Discussion by lsytj0413** [serverlessworkflow/specification#757] [Q&A] [answered]
As i know SWF doesn't support defines sub-state in another workflow state（which state only have actions which an performed sequential or parallel），How can i define an workflow like this：

![bpmn-example drawio](https://user-images.githubusercontent.com/6871552/220502368-6cbde36a-f459-42e8-852b-4813e2945b09.png)

NOTE:
- If we use subflow we can achieve this by defines echo composite states into another workflow，but this is complex when this subflow is only use by there（will not be reused）

**Comment by cdavernas** [comment, marked as answer]
@lsytj0413 This is now possible using the last versions of the DSL, which now supports composite tasks.
